src/engine/PDKind.cc

```diff
@@ -0,0 +1,295 @@
+﻿#include "PDKind.h"
+
+#include "Common.h"
+#include "MainSolver.h"
+#include "ModelBasedProjection.h"
+#include "PTRef.h"
+#include "QuantifierElimination.h"
+#include "TermUtils.h"
+#include "TransformationUtils.h"
+#include "transformers/BasicTransformationPipelines.h"
+#include "transformers/SingleLoopTransformation.h"
+#include <memory>
+#include <queue>
+#include <set>
+#include <tuple>
+
+VerificationResult PDKind::solve(ChcDirectedHyperGraph const & graph) {
+    auto pipeline = Transformations::towardsTransitionSystems();
+    auto transformationResult = pipeline.transform(std::make_unique<ChcDirectedHyperGraph>(graph));
+    auto transformedGraph = std::move(transformationResult.first);
+    auto translator = std::move(transformationResult.second);
+    if (transformedGraph->isNormalGraph()) {
+        auto normalGraph = transformedGraph->toNormalGraph();
+        auto res = solve(*normalGraph);
+        return computeWitness ? translator->translate(std::move(res)) : std::move(res);
+    }
+    return VerificationResult(VerificationAnswer::UNKNOWN);
+}
+
+VerificationResult PDKind::solve(ChcDirectedGraph const & system) {
+    if (isTrivial(system)) {
+        return solveTrivial(system);
+    }
+    if (isTransitionSystem(system)) {
+        auto ts = toTransitionSystem(system);
+        return solveTransitionSystem(*ts);
+    }
+    SingleLoopTransformation transformation;
+    auto[ts, backtranslator] = transformation.transform(system);
+    assert(ts);
+    auto res = solveTransitionSystem(*ts);
+    return res;
+}
+
+/* Create a wrapper structure which will hold the result and also an induction invariant or a number of steps to a counter example */
+VerificationResult PDKind::solveTransitionSystem (TransitionSystem const & system) {
+    PTRef init = system.getInit();
+    PTRef query = system.getQuery();
+    PTRef transition = system.getTransition();
+
+    ReachabilityChecker reachability_checker(logic, system);
+
+    SMTConfig config;
+    TimeMachine tm{logic};
+    MainSolver solver(logic, config, "PDKIND");
```

Comment:
This solver is not used, no?
You just insert `init` later, but otherwise, it is not accessed at all, no?
Seems like it can be removed.